app/models/concerns/billing/usage_support.rb

```diff
@@ -4,7 +4,7 @@ module Billing::UsageSupport
   def track_billing_usage(action, model: nil, count: 1)
     model ||= self.class
 
-    send(BulletTrain::Billing::Usage.parent_association).billing_usage_trackers.current.each do |tracker|
+    billing_usage_trackers.current.each do |tracker|
```

Comment:
I removed the `send(BulletTrain::Billing::Usage.parent_association)` because with these changes, `parent_association` needs to be dynamic, but being a class var makes it tricky. Removing it makes it work for our case, but it may not be ideal for the main gem.

Reply:
I'm not super familiar with this gem, so I'm not entirely sure, but I think this may cause trouble for apps that are already using this gem.

To take the [example from the docs](https://bullettrain.co/docs/billing/usage#tracking-usage) imagine that someone has done this:

```ruby
class Blogs::Post < ApplicationRecord
  # ...

  def publish!
    update(published_at: Time.zone.now)
    track_billing_usage(:published)
  end
end
```

Before this change the `track_billing_usage` call would be equivalent to doing:

```ruby
Billing::Usage::Trackers.where(team_id: self.team).current.each do |tracker|
  # ...
end
```

And after if would be doing:

```ruby
Billing::Usage::Trackers.where(trackable_type: "Blog", trackable_id: self.id).current.each do |tracker|
  # ...
end
```

It seems like the usage would start getting tracked in a different place and the counts would be thrown off. Do we need to figure out a way to migrate the existing records into a new place? Or is there some way to rebuild the counts from scratch after making the update?